packages/itwin/tree-widget/src/components/tree-header/TreeHeader.tsx

```diff
@@ -6,46 +6,76 @@
+import { ButtonGroup, Divider, DropdownMenu, IconButton, MenuItem, SearchBox } from "@itwin/itwinui-react";
+import { TreeWidget } from "../../TreeWidget";
 
 import type { Viewport } from "@itwin/core-frontend";
-import type { SearchBoxProps } from "./SearchBox";
+import type { CommonProps } from "@itwin/core-react";
 
 /** @internal */
 export interface TreeHeaderButtonProps {
   viewport: Viewport;
 }
 
 /** @internal */
-export interface TreeHeaderProps extends Omit<SearchBoxProps,
-| "onIconClick"
-| "valueChangedDelay"
-| "searchOpen"
-| "onSearchOpen"
-| "onSearchClose"
-> {
+export interface TreeHeaderProps extends CommonProps {
+  /** Filtering is cleared after everything's loaded */
+  onFilterStart: (newFilter: string) => void;
+  /** listens for onClick event for Clear (x) icon */
+  onFilterClear: () => void;
+  /** Total number of results/entries */
+  resultCount?: number;
+  /** Current selected result index */
+  selectedIndex?: number;
+  /** Callback to currently selected result/entry change */
+  onSelectedChanged: (index: number) => void;
   /** Header buttons */
   children?: React.ReactNode;
 }
 
 /** @internal */
 export function TreeHeader(props: TreeHeaderProps) {
-  const { children, ...restProps } = props;
-  const [searchOpen, setSearchOpen] = useState(false);
-
+  const { onFilterStart, onFilterClear, resultCount, selectedIndex, onSelectedChanged, children, className } = props;
+  const [isSearchOpen, setIsSearchOpen] = useState<boolean>(false);
   return (
-    <div className={classnames("tree-widget-search-bar", props.className)}>
-      <HeaderButtons contracted={searchOpen}>{children}</HeaderButtons>
-      <div className="search-bar-search-container">
-        <SearchBox
-          {...restProps}
-          searchOpen={searchOpen}
-          onSearchOpen={() => setSearchOpen(true)}
-          onSearchClose={() => setSearchOpen(false)}
-        />
-      </div>
+    <div className={classnames("tree-widget-tree-header", className)}>
+      <HeaderButtons contracted={isSearchOpen}>
+        {children}
+      </HeaderButtons>
+      <SearchBox
+        expandable
+        onExpand={() => setIsSearchOpen(true)}
+        onCollapse={() => setIsSearchOpen(false)}
+        className={classnames("tree-widget-search-bar", !isSearchOpen && "contracted")}
+      >
+        <SearchBox.CollapsedState>
+          <SearchBox.ExpandButton
+            title={TreeWidget.translate("searchBox.searchForSomething")}
+            aria-label={TreeWidget.translate("searchBox.open")}
+            size="small"
+          />
+        </SearchBox.CollapsedState>
+        <SearchBox.ExpandedState >
+          <SearchBox.Input
+            placeholder={TreeWidget.translate("searchBox.search")}
+            onChange={(e) => e.currentTarget.value ? onFilterStart(e.currentTarget.value) : onFilterClear()}
+            className="search-input"
+          />
+          {resultCount
```

Comment:
I think this check can be moved into `SearchResultStepper`

Reply:
fixed